src/server/project.ts

```diff
@@ -1780,6 +1788,46 @@ namespace ts.server {
         getIncompleteCompletionsCache() {
             return this.projectService.getIncompleteCompletionsCache();
         }
+
+        /*@internal*/
+        getNoDtsResolutionProject(rootFileNames: readonly string[]): Project {
+            if (!this.noDtsResolutionProject) {
+                const options: CompilerOptions = {
+                    ...this.getCompilerOptions(),
+                    noDtsResolution: true,
+                    allowJs: true,
+                    maxNodeModuleJsDepth: 3,
+                    diagnostics: false,
+                    skipLibCheck: true,
+                    sourceMap: false,
+                    types: ts.emptyArray,
+                    lib: ts.emptyArray,
+                    noLib: true,
+                };
+                this.noDtsResolutionProject = new AuxiliaryProject(this.projectService, this.documentRegistry, options);
+            }
+
+            enumerateInsertsAndDeletes<NormalizedPath, NormalizedPath>(
+                rootFileNames.map(toNormalizedPath),
+                this.noDtsResolutionProject.getRootFiles(),
+                compareStringsCaseInsensitive,
+                pathToAdd => {
+                    const info = this.noDtsResolutionProject!.getScriptInfo(pathToAdd);
+                    if (info) {
+                        this.noDtsResolutionProject!.addRoot(info, pathToAdd);
+                    }
+                },

```

Comment:
File change will cause it to schedule update graph.. You probably just dont want to schedule it and do it on demand instead.

Reply:
I’m not sure I understand—any time this runs, we are about to query the language service, so I need `updateGraph` to run after this batch of inserts/deletes.

Reply:
Sorry for not being clear..  I am not talking about this file add and delete. I mean that after this operation if say you change a file thats included in this project, editor services will schedule update graph.. you would want to change ScheduleUpdateGraph as noop so its not updating in background.. i forget if we are already doing it for auto import provider but it would benefit from that too.. (you still want to mark project dirty etc but not schedule things)
There is scheduling for invalidating failed lookup locations as well which can be deferred till needed. 

So https://github.com/microsoft/TypeScript/blob/main/src/server/editorServices.ts#L993 and https://github.com/microsoft/TypeScript/blob/main/src/server/project.ts#L527

Reply:
Ah, that makes sense. Will look into that, thanks.

Reply:
I’m not quite sure what to do with `scheduleInvalidateResolutionsOfFailedLookupLocations`. `updateGraphWorker` ultimately calls `resolutionCache.invalidateResolutionsOfFailedLookupLocations()`, so can I just make `AuxiliaryProject.scheduleInvalidateResolutionsOfFailedLookupLocations` a no-op, since the work it’s supposed to schedule seems to happen as part of `updateGraph`?

Reply:
Yes Thats the correct way to handle it..